File: week01/DAYE/1206.py
for test_case in range(1, 11):
    N = int(input())
    h = list(map(int, input().split()))
 
    answer = 0
 
    for i in range(2, N - 2):
        max_height = max(
            h[i - 2],
            h[i - 1],
            h[i + 1],
            h[i + 2]
        )
 
        if h[i] > max_height:
            answer += h[i] - max_height
 
    print(f"#{test_case} {answer}")

File: week01/DAYE/1206.java
import java.util.Scanner;
import java.io.FileInputStream;

class Solution
{
    public static void main(String args[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);
 
        for (int test_case = 1; test_case <= 10; test_case++) {
            int N = sc.nextInt();
            int[] h = new int[N];
            for (int i = 0; i < N; i++) {
                h[i] = sc.nextInt();
            }
            int answer = 0;
            for (int i = 2; i < N - 2; i++) {
                int max_height = Math.max(
                    Math.max(h[i - 2], h[i - 1]),
                    Math.max(h[i + 1], h[i + 2])
                );
                if (h[i] > max_height) {
                    answer += h[i] - max_height;
                }
            }
            System.out.println("#" + test_case + " " + answer);
        }
    }
}